### PR TITLE
feat:Delay tooltip presentations by 300ms

### PR DESCRIPTION
**Pull request by AdiAr11**
### Feature Preview

Will close issue #3310 

#### PR Checklist

- [x] My code adheres to the [AppFlowy Style Guide](https://appflowy.gitbook.io/docs/essential-documentation/contribute-to-appflowy/software-contributions/submitting-code/style-guides)
- [x] I've listed at least one issue that this PR fixes in the description above.
- [ ] I've added a test(s) to validate changes in this PR, or this PR only contains semantic changes.
- [x] All existing tests are passing.

### CODE REVIEW

**Comment by AdiAr11** [comment]
@Xazin @MayurSMahajan I have added waitDuration to Tooltip widgets. But in very few places, like more_button file, tooltip is being used like the following:
PopupMenuButton<int>(
      color: Theme.of(context).colorScheme.surfaceVariant,
      offset: const Offset(0, 30),
      tooltip: LocaleKeys.moreAction_moreOptions.tr(),
      child: FlowySvg(
         ...
      ),
      itemBuilder: (context) {}...
}
Under the hood, PopupMenuButton uses Tooltip widget, but since its a part of Flutter widget, can't push changes there.
One solution is to create another widget which would have a Stack widget, and wrap PopupMenuButton with that widget. What do you guys suggest?
Thank you

**Comment by Xazin** [comment]
> @Xazin @MayurSMahajan I have added waitDuration to Tooltip widgets. But in very few places, like more_button file, tooltip is being used like the following:
> 
> PopupMenuButton<int>(
> 
>       color: Theme.of(context).colorScheme.surfaceVariant,
> 
>       offset: const Offset(0, 30),
> 
>       tooltip: LocaleKeys.moreAction_moreOptions.tr(),
> 
>       child: FlowySvg(
> 
>          ...
> 
>       ),
> 
>       itemBuilder: (context) {}...
> 
> }
> 
> Under the hood, PopupMenuButton uses Tooltip widget, but since its a part of Flutter widget, can't push changes there.
> 
> One solution is to create another widget which would have a Stack widget, and wrap PopupMenuButton with that widget. What do you guys suggest?
> 
> Thank you

Don't use Stack, just wrap with Tooltip. 

Make an extension on Tooltip with a custom method eg. Tooltip.delayed, that returns a tooltip with the default delay used in AppFlowy.

**Comment by AdiAr11** [comment]
@Xazin, Yes, I tried wrapping with Tooltip widget but that did not work either.

**Comment by Xazin** [comment]
> @Xazin, Yes, I tried wrapping with Tooltip widget but that did not work either.

If you wrap FlowySvg and not PopupMenuButton.

**Comment by AdiAr11** [comment]
> > @Xazin, Yes, I tried wrapping with Tooltip widget but that did not work either.
> 
> If you wrap FlowySvg and not PopupMenuButton.

That I did not try yet. Thank you!

**Comment by AdiAr11** [comment]
@Xazin I have added the waitDuration on Tooltips. Please review.

File: frontend/scripts/install_dev_env/install_macos.sh [hunk 1]
```diff
@@ -41,9 +41,9 @@ printMessage "Setting up Flutter"
 
 # Get the current Flutter version
 FLUTTER_VERSION=$(flutter --version | grep -oE 'Flutter [^ ]+' | grep -oE '[^ ]+$')
-# Check if the current version is 3.10.1
-if [ "$FLUTTER_VERSION" = "3.10.1" ]; then
-   echo "Flutter version is already 3.10.1"
+# Check if the current version is 3.10.6
+if [ "$FLUTTER_VERSION" = "3.10.6" ]; then
```
Comment (Xazin) [inline frontend/scripts/install_dev_env/install_macos.sh:45]: Please don't update the flutter version.

**Review by MayurSMahajan — APPROVED** [review]

**Comment by Xazin** [comment]
I am not a huge fan of this solution, I will add a commit to show you how to do this.

**Comment by AdiAr11** [comment]
@Xazin. Yes please do. Although, I think i know the solution you are looking for. Do you suggest i create a custom class and provide it default value of waitDuration and replace Tooltip by this class?

File: frontend/scripts/install_dev_env/install_macos.sh (hunk 1 above)
Comment (AdiAr11) [reply frontend/scripts/install_dev_env/install_macos.sh:45]: I'm really sorry, I don't know how this got pushed. Will revert this

**Comment by AdiAr11** [comment]
@Xazin I just saw your changes. Thank you so much for your help!!

**Comment by Xazin** [comment]
> @Xazin. Yes please do. Although, I think i know the solution you are looking for. Do you suggest i create a custom class and provide it default value of waitDuration and replace Tooltip by this class?
Yes, actually the class already existed, it just had to be moved to the infra package.

**Comment by LucasXu0** [comment]
the name(FlowyTooltip.delayedTooltip) of the constructor seems a bit redundant. How about using FlowyTooltip.delayed instead?

**Comment by AdiAr11** [comment]
@Xazin , should i change it to FlowyTooltip.delayed?

**Comment by Xazin** [comment]
> @Xazin , should i change it to FlowyTooltip.delayed?

Yeah sounds good 👍

**Review by Xazin — APPROVED** [review]

**Pull request merged**